### ISSUE TITLE
reconcile failed,  "error": "Job.batch \"volsync-dst-my-backup-backup-destination\" is invalid: spec.template: Invalid value:, field is immutable, 

### ISSUE THREAD

**Issue by v-harihar** [backube/volsync#310]
**Describe the bug**

This happens during the restore process from backup taken on restic repo to pvc and got produced while working on https://github.com/backube/volsync/pull/208. The replication destination is created using following values https://github.com/backube/volsync/blob/32a2a20d7e61a8296a59ad735e092eea0bfc189c/kubectl-volsync/cmd/pvbackup_restore.go#L302-L327  **Operator logs**
```
1.6572717338878305e+09	DEBUG	controllers.ReplicationDestination	transitioning to synchronization state	{"replicationdestination": "dest/my-backup-backup-destination"}
1.6572717338939772e+09	INFO	controllers.ReplicationDestination	allocating cache volume	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "PVC": "volsync-my-backup-backup-destination-cache"}
1.6572717338969615e+09	DEBUG	controllers.ReplicationDestination	PVC reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "PVC": "volsync-my-backup-backup-destination-cache", "operation": "created"}
1.6572717338972464e+09	DEBUG	events	Normal	{"object": {"kind":"ReplicationDestination","namespace":"dest","name":"my-backup-backup-destination","uid":"263432bc-0b0a-4949-bbaf-d995b4d73a82","apiVersion":"volsync.backube/v1alpha1","resourceVersion":"40308"}, "reason": "PersistentVolumeClaimCreated", "message": "created PersistentVolumeClaim/volsync-my-backup-backup-destination-cache to receive incoming data"}
1.657271733899673e+09	DEBUG	controllers.ReplicationDestination	ServiceAccount reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "ServiceAccount": "dest/volsync-dst-my-backup-backup-destination", "operation": "created"}
1.657271733904202e+09	DEBUG	controllers.ReplicationDestination	Role reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "Role": "dest/volsync-dst-my-backup-backup-destination", "operation": "created"}
1.6572717339074318e+09	DEBUG	controllers.ReplicationDestination	RoleBinding reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "RoleBinding": "dest/volsync-dst-my-backup-backup-destination", "operation": "created"}
1.6572717339075124e+09	INFO	controllers.ReplicationDestination	job actions	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "job": "dest/volsync-dst-my-backup-backup-destination", "actions": ["restore"]}
1.6572717339186459e+09	INFO	controllers.ReplicationDestination	allocating cache volume	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "PVC": "volsync-my-backup-backup-destination-cache"}
1.6572717339187958e+09	DEBUG	controllers.ReplicationDestination	PVC reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "PVC": "volsync-my-backup-backup-destination-cache", "operation": "unchanged"}
1.6572717339188778e+09	DEBUG	controllers.ReplicationDestination	ServiceAccount reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "ServiceAccount": "dest/volsync-dst-my-backup-backup-destination", "operation": "unchanged"}
1.6572717339189603e+09	DEBUG	controllers.ReplicationDestination	Role reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "Role": "dest/volsync-dst-my-backup-backup-destination", "operation": "unchanged"}
1.6572717339190392e+09	DEBUG	controllers.ReplicationDestination	RoleBinding reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "RoleBinding": "dest/volsync-dst-my-backup-backup-destination", "operation": "unchanged"}
1.6572717339191086e+09	INFO	controllers.ReplicationDestination	job actions	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "job": "dest/volsync-dst-my-backup-backup-destination", "actions": ["restore"]}
1.6572717339346516e+09	INFO	controllers.ReplicationDestination	allocating cache volume	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "PVC": "volsync-my-backup-backup-destination-cache"}
1.6572717339348109e+09	DEBUG	controllers.ReplicationDestination	PVC reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "PVC": "volsync-my-backup-backup-destination-cache", "operation": "unchanged"}
1.6572717339348772e+09	DEBUG	controllers.ReplicationDestination	ServiceAccount reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "ServiceAccount": "dest/volsync-dst-my-backup-backup-destination", "operation": "unchanged"}
1.657271733934972e+09	DEBUG	controllers.ReplicationDestination	Role reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "Role": "dest/volsync-dst-my-backup-backup-destination", "operation": "unchanged"}
1.657271733935036e+09	DEBUG	controllers.ReplicationDestination	RoleBinding reconciled	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "RoleBinding": "dest/volsync-dst-my-backup-backup-destination", "operation": "unchanged"}
1.657271733935096e+09	INFO	controllers.ReplicationDestination	job actions	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "job": "dest/volsync-dst-my-backup-backup-destination", "actions": ["restore"]}
1.657271733941258e+09	ERROR	controllers.ReplicationDestination	reconcile failed	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "job": "dest/volsync-dst-my-backup-backup-destination", "error": "Job.batch \"volsync-dst-my-backup-backup-destination\" is invalid: spec.template: Invalid value: core.PodTemplateSpec{ObjectMeta:v1.ObjectMeta{Name:\"volsync-dst-my-backup-backup-destination\", GenerateName:\"\", Namespace:\"\", SelfLink:\"\", UID:\"\", ResourceVersion:\"\", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{\"app.kubernetes.io/created-by\":\"volsync\", \"controller-uid\":\"fbff731a-0f0d-401a-a0d2-bcd7dbf9901b\", \"job-name\":\"volsync-dst-my-backup-backup-destination\"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ZZZ_DeprecatedClusterName:\"\", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:core.PodSpec{Volumes:[]core.Volume{core.Volume{Name:\"data\", VolumeSource:core.VolumeSource{HostPath:(*core.HostPathVolumeSource)(nil), EmptyDir:(*core.EmptyDirVolumeSource)(nil), GCEPersistentDisk:(*core.GCEPersistentDiskVolumeSource)(nil), AWSElasticBlockStore:(*core.AWSElasticBlockStoreVolumeSource)(nil), GitRepo:(*core.GitRepoVolumeSource)(nil), Secret:(*core.SecretVolumeSource)(nil), NFS:(*core.NFSVolumeSource)(nil), ISCSI:(*core.ISCSIVolumeSource)(nil), Glusterfs:(*core.GlusterfsVolumeSource)(nil), PersistentVolumeClaim:(*core.PersistentVolumeClaimVolumeSource)(0xc0163893b0), RBD:(*core.RBDVolumeSource)(nil), Quobyte:(*core.QuobyteVolumeSource)(nil), FlexVolume:(*core.FlexVolumeSource)(nil), Cinder:(*core.CinderVolumeSource)(nil), CephFS:(*core.CephFSVolumeSource)(nil), Flocker:(*core.FlockerVolumeSource)(nil), DownwardAPI:(*core.DownwardAPIVolumeSource)(nil), FC:(*core.FCVolumeSource)(nil), AzureFile:(*core.AzureFileVolumeSource)(nil), ConfigMap:(*core.ConfigMapVolumeSource)(nil), VsphereVolume:(*core.VsphereVirtualDiskVolumeSource)(nil), AzureDisk:(*core.AzureDiskVolumeSource)(nil), PhotonPersistentDisk:(*core.PhotonPersistentDiskVolumeSource)(nil), Projected:(*core.ProjectedVolumeSource)(nil), PortworxVolume:(*core.PortworxVolumeSource)(nil), ScaleIO:(*core.ScaleIOVolumeSource)(nil), StorageOS:(*core.StorageOSVolumeSource)(nil), CSI:(*core.CSIVolumeSource)(nil), Ephemeral:(*core.EphemeralVolumeSource)(nil)}}, core.Volume{Name:\"cache\", VolumeSource:core.VolumeSource{HostPath:(*core.HostPathVolumeSource)(nil), EmptyDir:(*core.EmptyDirVolumeSource)(nil), GCEPersistentDisk:(*core.GCEPersistentDiskVolumeSource)(nil), AWSElasticBlockStore:(*core.AWSElasticBlockStoreVolumeSource)(nil), GitRepo:(*core.GitRepoVolumeSource)(nil), Secret:(*core.SecretVolumeSource)(nil), NFS:(*core.NFSVolumeSource)(nil), ISCSI:(*core.ISCSIVolumeSource)(nil), Glusterfs:(*core.GlusterfsVolumeSource)(nil), PersistentVolumeClaim:(*core.PersistentVolumeClaimVolumeSource)(0xc0163893c8), RBD:(*core.RBDVolumeSource)(nil), Quobyte:(*core.QuobyteVolumeSource)(nil), FlexVolume:(*core.FlexVolumeSource)(nil), Cinder:(*core.CinderVolumeSource)(nil), CephFS:(*core.CephFSVolumeSource)(nil), Flocker:(*core.FlockerVolumeSource)(nil), DownwardAPI:(*core.DownwardAPIVolumeSource)(nil), FC:(*core.FCVolumeSource)(nil), AzureFile:(*core.AzureFileVolumeSource)(nil), ConfigMap:(*core.ConfigMapVolumeSource)(nil), VsphereVolume:(*core.VsphereVirtualDiskVolumeSource)(nil), AzureDisk:(*core.AzureDiskVolumeSource)(nil), PhotonPersistentDisk:(*core.PhotonPersistentDiskVolumeSource)(nil), Projected:(*core.ProjectedVolumeSource)(nil), PortworxVolume:(*core.PortworxVolumeSource)(nil), ScaleIO:(*core.ScaleIOVolumeSource)(nil), StorageOS:(*core.StorageOSVolumeSource)(nil), CSI:(*core.CSIVolumeSource)(nil), Ephemeral:(*core.EphemeralVolumeSource)(nil)}}}, InitContainers:[]core.Container(nil), Containers:[]core.Container{core.Container{Name:\"restic\", Image:\"quay.io/backube/volsync-mover-restic:latest\", Command:[]string{\"/entry.sh\"}, Args:[]string{\"restore\"}, WorkingDir:\"\", Ports:[]core.ContainerPort(nil), EnvFrom:[]core.EnvFromSource(nil), Env:[]core.EnvVar{core.EnvVar{Name:\"FORGET_OPTIONS\", Value:\"--keep-last 1\", ValueFrom:(*core.EnvVarSource)(nil)}, core.EnvVar{Name:\"DATA_DIR\", Value:\"/data\", ValueFrom:(*core.EnvVarSource)(nil)}, core.EnvVar{Name:\"RESTIC_CACHE_DIR\", Value:\"/cache\", ValueFrom:(*core.EnvVarSource)(nil)}, core.EnvVar{Name:\"RESTORE_AS_OF\", Value:\"\", ValueFrom:(*core.EnvVarSource)(nil)}, core.EnvVar{Name:\"SELECT_PREVIOUS\", Value:\"0\", ValueFrom:(*core.EnvVarSource)(nil)}, core.EnvVar{Name:\"RESTIC_REPOSITORY\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06700)}, core.EnvVar{Name:\"RESTIC_PASSWORD\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06720)}, core.EnvVar{Name:\"AWS_ACCESS_KEY_ID\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06740)}, core.EnvVar{Name:\"AWS_SECRET_ACCESS_KEY\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06760)}, core.EnvVar{Name:\"AWS_DEFAULT_REGION\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06780)}, core.EnvVar{Name:\"ST_AUTH\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f067a0)}, core.EnvVar{Name:\"ST_USER\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f067c0)}, core.EnvVar{Name:\"ST_KEY\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f067e0)}, core.EnvVar{Name:\"OS_AUTH_URL\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06800)}, core.EnvVar{Name:\"OS_REGION_NAME\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06820)}, core.EnvVar{Name:\"OS_USERNAME\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06840)}, core.EnvVar{Name:\"OS_USER_ID\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06860)}, core.EnvVar{Name:\"OS_PASSWORD\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f068a0)}, core.EnvVar{Name:\"OS_TENANT_ID\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f068c0)}, core.EnvVar{Name:\"OS_TENANT_NAME\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f068e0)}, core.EnvVar{Name:\"OS_USER_DOMAIN_NAME\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06900)}, core.EnvVar{Name:\"OS_USER_DOMAIN_ID\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06920)}, core.EnvVar{Name:\"OS_PROJECT_NAME\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06940)}, core.EnvVar{Name:\"OS_PROJECT_DOMAIN_NAME\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06960)}, core.EnvVar{Name:\"OS_PROJECT_DOMAIN_ID\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06980)}, core.EnvVar{Name:\"OS_TRUST_ID\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f069a0)}, core.EnvVar{Name:\"OS_APPLICATION_CREDENTIAL_ID\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f069c0)}, core.EnvVar{Name:\"OS_APPLICATION_CREDENTIAL_NAME\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f069e0)}, core.EnvVar{Name:\"OS_APPLICATION_CREDENTIAL_SECRET\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06a00)}, core.EnvVar{Name:\"OS_STORAGE_URL\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06a20)}, core.EnvVar{Name:\"OS_AUTH_TOKEN\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06a40)}, core.EnvVar{Name:\"B2_ACCOUNT_ID\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06a60)}, core.EnvVar{Name:\"B2_ACCOUNT_KEY\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06a80)}, core.EnvVar{Name:\"AZURE_ACCOUNT_NAME\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06aa0)}, core.EnvVar{Name:\"AZURE_ACCOUNT_KEY\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06ac0)}, core.EnvVar{Name:\"GOOGLE_PROJECT_ID\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06b00)}, core.EnvVar{Name:\"GOOGLE_APPLICATION_CREDENTIALS\", Value:\"\", ValueFrom:(*core.EnvVarSource)(0xc006f06b20)}}, Resources:core.ResourceRequirements{Limits:core.ResourceList(nil), Requests:core.ResourceList(nil)}, VolumeMounts:[]core.VolumeMount{core.VolumeMount{Name:\"data\", ReadOnly:false, MountPath:\"/data\", SubPath:\"\", MountPropagation:(*core.MountPropagationMode)(nil), SubPathExpr:\"\"}, core.VolumeMount{Name:\"cache\", ReadOnly:false, MountPath:\"/cache\", SubPath:\"\", MountPropagation:(*core.MountPropagationMode)(nil), SubPathExpr:\"\"}}, VolumeDevices:[]core.VolumeDevice(nil), LivenessProbe:(*core.Probe)(nil), ReadinessProbe:(*core.Probe)(nil), StartupProbe:(*core.Probe)(nil), Lifecycle:(*core.Lifecycle)(nil), TerminationMessagePath:\"/dev/termination-log\", TerminationMessagePolicy:\"File\", ImagePullPolicy:\"Always\", SecurityContext:(*core.SecurityContext)(0xc00a37d3e0), Stdin:false, StdinOnce:false, TTY:false}}, EphemeralContainers:[]core.EphemeralContainer(nil), RestartPolicy:\"Never\", TerminationGracePeriodSeconds:(*int64)(0xc00c27ca70), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:\"ClusterFirst\", NodeSelector:map[string]string(nil), ServiceAccountName:\"volsync-dst-my-backup-backup-destination\", AutomountServiceAccountToken:(*bool)(nil), NodeName:\"kind-control-plane\", SecurityContext:(*core.PodSecurityContext)(0xc005250a80), ImagePullSecrets:[]core.LocalObjectReference(nil), Hostname:\"\", Subdomain:\"\", SetHostnameAsFQDN:(*bool)(nil), Affinity:(*core.Affinity)(nil), SchedulerName:\"default-scheduler\", Tolerations:[]core.Toleration{core.Toleration{Key:\"node.kubernetes.io/not-ready\", Operator:\"Exists\", Value:\"\", Effect:\"NoExecute\", TolerationSeconds:(*int64)(0xc00c27ca78)}, core.Toleration{Key:\"node.kubernetes.io/unreachable\", Operator:\"Exists\", Value:\"\", Effect:\"NoExecute\", TolerationSeconds:(*int64)(0xc00c27ca80)}}, HostAliases:[]core.HostAlias(nil), PriorityClassName:\"\", Priority:(*int32)(nil), PreemptionPolicy:(*core.PreemptionPolicy)(nil), DNSConfig:(*core.PodDNSConfig)(nil), ReadinessGates:[]core.PodReadinessGate(nil), RuntimeClassName:(*string)(nil), Overhead:core.ResourceList(nil), EnableServiceLinks:(*bool)(nil), TopologySpreadConstraints:[]core.TopologySpreadConstraint(nil), OS:(*core.PodOS)(nil)}}: field is immutable"}
github.com/backube/volsync/controllers/mover/restic.(*Mover).Synchronize
	/home/vharihar/project/volsync/controllers/mover/restic/mover.go:118
github.com/backube/volsync/controllers.(*rdMachine).Synchronize
	/home/vharihar/project/volsync/controllers/replicationdestination_controller.go:261
github.com/backube/volsync/controllers/statemachine.doSynchronizingState
	/home/vharihar/project/volsync/controllers/statemachine/machine.go:102
github.com/backube/volsync/controllers/statemachine.Run
	/home/vharihar/project/volsync/controllers/statemachine/machine.go:70
github.com/backube/volsync/controllers.(*ReplicationDestinationReconciler).Reconcile
	/home/vharihar/project/volsync/controllers/replicationdestination_controller.go:135
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/usr/local/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:121
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/usr/local/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:320
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/usr/local/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/usr/local/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234

```
**Steps to reproduce**
The issue got produced during restore and during the developement of https://github.com/backube/volsync/pull/208
1. Run `minio server /data` (restic repo)
2. ./bin/kubectl-volsync pv-backup create --relationship pvb1 --pvcname src/pvc1 --restic-config restic-conf.toml --name my-backup --cronspec "*/2 * * * *"
3. ./bin/kubectl-volsync pv-backup restore --pvcname dest1/pvc1 --relationship pvr5 --repository my-backup --restic-config restic-conf.toml --capacity 1Gi

sample restic.conf.toml
```

# cat restic-conf.toml 
# ****** Launch local s3 server. On Linux, execute below commands ****** 
# cmd ex:
# # wget https://dl.min.io/server/minio/release/linux-amd64/minio
# # chmod +x minio
# # ./minio server /data
#   API: http://10.74.17.163:9000  http://192.168.1.8:9000  http://172.17.0.1:9000  http://172.19.0.1:9000  http://192.168.122.1:9000      http://127.0.0.1:9000
#   RootUser: minioadmin
#   RootPass: minioadmin
# Open any of API links, enter the creds above and create a bucket
# for more info: https://docs.min.io/docs/minio-quickstart-guide.html
# **********************************************************************

# The repository url. Below URL is generated by running minio server locally as explained above
RESTIC_REPOSITORY = "s3:http://192.168.1.8:9000/restic-repo"
# The repository encryption key
RESTIC_PASSWORD = "my-secure-restic-password"
# Below creds are generated by running minio server locally as exaplained above
AWS_ACCESS_KEY_ID = "minioadmin"
AWS_SECRET_ACCESS_KEY = "minioadmin"
```

**Comment by JohnStrunk** [comment]
I think this is #302. Does it still occur w/ latest build from main?

**Comment by v-harihar** [comment]
With the latest, reported problem is not occuring.

Just want to understand the following behavior. Why/How replicationDestination gets modified ?

```
1.6572856863272593e+09	ERROR	controllers.ReplicationDestination	reconcile failed	{"replicationdestination": "dest/my-backup-backup-destination", "method": "Restic", "job": "dest/volsync-dst-my-backup-backup-destination", "error": "Operation cannot be fulfilled on jobs.batch \"volsync-dst-my-backup-backup-destination\": the object has been modified; please apply your changes to the latest version and try again"}
github.com/backube/volsync/controllers/mover/restic.(*Mover).Synchronize
	/home/vharihar/project/volsync/controllers/mover/restic/mover.go:117
github.com/backube/volsync/controllers.(*rdMachine).Synchronize
	/home/vharihar/project/volsync/controllers/replicationdestination_controller.go:264
github.com/backube/volsync/controllers/statemachine.doSynchronizingState
	/home/vharihar/project/volsync/controllers/statemachine/machine.go:102
github.com/backube/volsync/controllers/statemachine.Run
	/home/vharihar/project/volsync/controllers/statemachine/machine.go:70
github.com/backube/volsync/controllers.(*ReplicationDestinationReconciler).Reconcile
	/home/vharihar/project/volsync/controllers/replicationdestination_controller.go:135
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/usr/local/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:121
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/usr/local/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:320
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/usr/local/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/usr/local/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:234
1.657285686329961e+09	ERROR	Reconciler error	{"controller": "replicationdestination", "controllerGroup": "volsync.backube", "controllerKind": "ReplicationDestination", "replicationDestination": {"name":"my-backup-backup-destination","namespace":"dest"}, "namespace": "dest", "name": "my-backup-backup-destination", "reconcileID": "6900b91a-269b-4d1f-99b1-b4eefef64df2", "error": "Operation cannot be fulfilled on jobs.batch \"volsync-dst-my-backup-backup-destination\": the object has been modified; please apply your changes to the latest version and try again"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem

```

**Comment by JohnStrunk** [comment]
The problem fixed in 302 is an issue I introduced when I added the mover affinity code. It caused parts of the Job object to be modified (for affinity purposes), and it turned out those fields are immutable, hence the error.

What you're seeing above is because of the operator's object cache. The operator's client library caches objects to avoid doing frequent requests to the api server. Unfortunately, that means that when we try to update an object, that can fail because updates are predicated on a particular resource version, and the cached one is out of date, so the server rejects it.
The reconciler logs the above messages, but it retries, and that eventually refreshes the cache, allowing the update to succeed.
